packages/flutter/test/material/theme_data_test.dart

```diff
@@ -170,4 +170,138 @@ void main() {
     expect(theme.applyElevationOverlayColor, isTrue);
   });
 
+  testWidgets('ThemeData.copyWith return appropriate value', (WidgetTester tester) async {
```

Comment:
Thanks for the review.
I'm not sure about how to test **the argument type correctness**. Do you know where I can find an example of test of this kind please ?
I'm very surprised that analyzer can't detect the argument type error because of the fallback value.

```dart
// analyzer detects the type error
List<int> getList(String arg) => arg ;

// valid for the analyzer
List<int> getList(String arg) => arg ?? [0];
```

Isn't it possible to change the analyzer configuration to avoid this behavior ?

Reply:
That is indeed strange, I hadn't considered that the analyzer should've caught this problem to begin with. This might actually be a Dart analyzer bug... could you [file an issue](https://github.com/dart-lang/sdk/issues/new)? 

We could just move forward with the test that you've already created, since that would catch the issue that we ran into, but also catch regressions in copyWith's behavior. @HansMuller, what do you think?

Reply:
The new test looks OK to me.

This sounds like either an analyzer issue or something subtle about Dart that we don't understand. In this case, neither return value is a List<int>:

```dart
List<int> getList(String arg) => arg ?? [0];
```

I think the `[0]` is a List<dynamic>, `<int>[0]` would be a List<int>. 

Filing a Dart analyzer issue sounds like a good idea.

Reply:
It's a known issue : https://github.com/dart-lang/sdk/issues/25368, https://github.com/dart-lang/sdk/issues/36964

For now disabling `implicit-cast` seems the only option but in near future implicit casts will be disabled to migrate to NNBD cf. https://github.com/dart-lang/sdk/issues/36964#issuecomment-492249823